transformations/sentiment_emoji_augmenter/transformation.py

```diff
@@ -0,0 +1,115 @@
+import random
+
+from interfaces.SentenceOperation import SentenceAndTargetOperation
+import spacy
+from tasks.TaskTypes import TaskType
+
+"""
+Adds a positive labelled emoji as well as a positive emoteicon for positive sentences and vice versa. 
+And neutral smiley for unlabelled and neutral sentences.
+Since IMDB has labels +1 --> str(target) in ["1", "pos", "positive"] is used to determine if it's positive.
+"""
+
+emoji = {  # (facial expression, sentiment)-keys
+    ("love", +1.00): ["❤️", "💜", "💚", "💙", "💛", "💕"],
+    ("grin", +1.00): ["😀", "😄", "😃", "😆", "😅", "😂", "😁", "😻", "😍", "😈", "👌"],
+    ("taunt", +0.75): ["😛", "😝", "😜", "😋", "😇"],
+    ("smile", +0.50): ["😊", "😌", "😏", "😎", "☺", "👍"],
+    ("wink", +0.25): ["😉"],
+    ("blank", +0.00): ["😐", "😶"],
+    ("gasp", -0.05): ["😳", "😮", "😯", "😧", "😦", "🙀"],
+    ("worry", -0.25): ["😕", "😬"],
+    ("frown", -0.75): ["😟", "😒", "😔", "😞", "😠", "😩", "😫", "😡", "👿"],
+    ("cry", -1.00): ["😢", "😥", "😓", "😪", "😭", "😿"],
+}
+
+emoticons = {  # (facial expression, sentiment)-keys
```

Comment:
This seems somewhat arbitrary, can you add a justification for this splitting?

Reply:
This split comes from the library https://github.com/clips/pattern which was used as a source for these. And I think theoretically emojis are images while emoticons are keyboard characters.